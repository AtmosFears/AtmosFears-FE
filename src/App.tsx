import { useState } from 'react';

import './App.scss';
import MapComponent from './components/Map/MapComponent';
import TimeSeriesComponent from './components/TimeSeries/TimeSeriesComponent';

function App() {
  const [displayTimeSeries, setDisplayTimeSeries] = useState(false);

  let displayedComponent = <MapComponent />;
  if (displayTimeSeries) {
    displayedComponent = <TimeSeriesComponent />;
  }

  const switchView = () => {
    setDisplayTimeSeries(!displayTimeSeries);
  };

  return (
    <div className='App'>
      <button onClick={switchView}>Toggle view</button>
      {displayedComponent}
    </div>
  );
}

export default App;
