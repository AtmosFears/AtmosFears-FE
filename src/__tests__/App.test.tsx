import { cleanup, render, screen } from '@testing-library/react';

import App from '../Root';

// Example tests
describe('App', () => {
  afterEach(cleanup);

  it('snapshot', () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });

  it('renders a button with correct text', () => {
    render(<App />);

    const button = screen.queryByRole('button');

    expect(button).toBeInTheDocument();
    expect(button).toHaveTextContent('count is 0');
  });
});
