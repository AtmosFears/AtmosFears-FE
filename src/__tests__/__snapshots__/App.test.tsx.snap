// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > snapshot 1`] = `
<div>
  <div
    class="App"
  >
    <div>
      <a
        href="https://vitejs.dev"
        rel="noreferrer"
        target="_blank"
      >
        <img
          alt="Vite logo"
          class="logo"
          src="/@fs/assets/vite.svg"
        />
      </a>
      <a
        href="https://reactjs.org"
        rel="noreferrer"
        target="_blank"
      >
        <img
          alt="React logo"
          class="logo react"
          src="/src/assets/react.svg"
        />
      </a>
    </div>
    <h1>
      AQM4UA
    </h1>
    <div
      class="card"
    >
      <button>
        count is 
        0
      </button>
      <p>
        Edit 
        <code>
          src/App.tsx
        </code>
         and save to test HMR
      </p>
    </div>
    <p
      class="read-the-docs"
    >
      Click on the Vite and React logos to learn more
    </p>
  </div>
</div>
`;
